libraries/botbuilder-lg/src/analyzer.ts

```diff
@@ -236,7 +255,10 @@ export class Analyzer
                 this.templateMap[templateName].parameters.length === 0
             ) {
                 result.union(this.analyzeTemplate(templateName));
-            } else if (!result.TemplateReferences.includes(templateName)) {
+            } else if (
+                (this._analyzerOptions && this._analyzerOptions.ThrowOnRecursive === true) ||
```

Comment:
why use `this._analyzerOptions.ThrowOnRecursive === true` and not `this._analyzerOptions.ThrowOnRecursive`?

Reply:
Isn't `this._analyzerOptions.ThrowOnRecursive` be true as long as it's different from null?

Reply:
and false

Reply:
I think the `===` is better here to ensure the value is true and nothing else.